assets/js/wc-gateway-ppec-generate-cart.js

```diff
@@ -30,52 +30,66 @@
 			$( '#woo_pp_ec_button_product > *' ).css( 'pointer-events', 'none' );
 		} );
 
+	// True if the product is simple or the user selected a valid variation. False on variable product without a valid variation selected
+	var variation_valid = true;
+
+	// True if all the fields of the product form are valid. False otherwise
```

Comment:
Probably useful to add to this comment that this came as support for Product Addons, which adds form fields to the cart.